ghc-8.0 branch failed to build with ghc-8.0.1 (lts-7.9)

Currently, the ghc-8.0 branch failed to build with ghc-8.0.1 (lts-7.9), both the latest code in ghc-8.0 branch or the nightly snapshot (ghc-8.0-20170616.tar.gz) downloaded from http://ghcjs.luite.com/.

The error message:

~~~
ghcjs-0.2.1\src\Gen2\Linker.hs:354:42: error:
    * Couldn't match type `[Char]' with `ByteString'
      Expected type: ByteString
        Actual type: FilePath
    * In the first argument of `bsLit', namely
        `(nodeProgram nodeSettings)'
      In the expression: bsLit (nodeProgram nodeSettings)
      In the expression: ("node", bsLit (nodeProgram nodeSettings))
~~~

--- a/src/Gen2/Linker.hs
+++ b/src/Gen2/Linker.hs
@@ -47,6 +47,7 @@ import           Data.Array
 import           Data.Binary
 import           Data.ByteString          (ByteString)
 import qualified Data.ByteString          as B
+import qualified Data.ByteString.Char8    as BC
 import qualified Data.ByteString.Lazy     as BL
 import           Data.Char                (toLower, chr)
 import           Data.Function            (on)
@@ -351,7 +352,7 @@ writeRunner settings dflags out = when (gsBuildRunner settings) $ do
   T.writeFile runnerSrc $
     substPatterns [] [ ("js",     bsLit src)
                      , ("jsSize", T.pack (show $ B.length src))
-                     , ("node",   bsLit (nodeProgram nodeSettings))
+                     , ("node",   bsLit (BC.pack $ nodeProgram nodeSettings))
                      ] templ
   SysTools.runCc dflags [ Option "-o"
                         , FileOption "" runner

